### DISCUSSION TITLE
Both OneToOne and OneToMany relations between 2 entities leads to MetadataError

### DISCUSSION THREAD

**Discussion by noftaly** [mikro-orm/mikro-orm#2205] [Q&A] [answered]
Hello there!
I'm pretty new to SQL, and I'm experimenting Mikro-ORM with Postgres. I'm using Nest.js and the TsMorph Metadata Provider.

I have two entities, `Application` holding `Content`s, defined as following.
Each application holds many contents through `.contents`, but they also have 1 special content, stored in `.cover`.
It worked well until I added this "cover" mechanism.

```ts
@Entity()
export class Application {
  @PrimaryKey()
  applicationId: string = nanoid(10);

  @Property()
  name!: string;

  @OneToOne('Content', 'application', { owner: true, orphanRemoval: true })
  /**
   * Also tried with `inversedBy: 'application'`, and with those 2 options with only the option object, not
   * the `'Content', 'application', ` parameters
   */
  cover?: Content;

  @OneToMany('Content', 'application', { cascade: [Cascade.REMOVE] })
  contents = new Collection<Content>(this);

  constructor(name: string) {
    this.name = name;
  }
}

@Entity()
export class Content {
  @PrimaryKey()
  contentId: string = nanoid(10);

  @Property()
  name!: string;

  @ManyToOne()
  application!: Application;

  constructor(application: Application, name: string) {
    this.application = application;
    this.name = name;
  }
}
```

Now here are my questions:
- I get this error `MetadataError: Both Application.cover and Content.application are defined as owning sides, use 'mappedBy' on one of them`. As I told you I'm pretty new to sql, so I don't know what I have to do, I've tried putting `mappedBy`, `inversedBy` and `owner` on the App's 1t1 options, but I always get this error.
I have a feeling I should add the `mappedBy` to Content's MtM, but it is not possible.
Are my relationship messed up? How can I solve this?
- I managed to define these two entities without circular dependencies, but I now need to fully import each entity in the other's file. Is this a problem?

Thanks for your precious help!

**Comment by B4nan** [comment, marked as answer]
1. You can't map two inverse sides to a single owning side (which is exactly what you are doing). 
2. For bidirectional 1:1 relation, both sides need to be 1:1, while you are trying to have m:1 on the other side.

I am not sure what exactly are you up to, but this is not technically possible. What you can do is to have unidirectional 1:1 - in that case just drop the `mappedBy` parameter from it.

```ts
@OneToOne({ orphanRemoval: true })
cover?: Content;
```

This will end up with additional FK on the `Application` side

**Reply by noftaly** [reply]
Thanks! This solved my problem, although I've approach my issue from another angle now and don't have this situation anymore.
